src/app/cake/cake.service.ts

```diff
@@ -0,0 +1,29 @@
+import { Connection } from 'mongoose';
+
+import { DBService } from '../../components/database-service.component';
+import { CakeModel } from '../cake/cake.model';
+import { ICake, ICakeQuery } from '../cake/cake.interface';
+
+export class CakeService {
+    private dbService: DBService<ICake, ICakeQuery>;
+
+    constructor(conn: Connection) {
+        this.dbService = new DBService(conn, 'Cake', CakeModel);
```

Comment:
You can define the cake `Cake` hard coded value in the `cake.model.ts` file as `modelName` 😅